cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PullCommand.java

```diff
@@ -163,6 +179,87 @@ public void execute() {
             }
         }
 
+        Settings settings;
+        try {
+            settings = RepoUtils.readSettings();
+        } catch (SettingsTomlException e) {
+            settings = Settings.from();
+        }
+
+        Repository targetRepository = null;
+        if (repositoryName != null) {
+            for (Repository repository : settings.getRepositories()) {
+                if (repository.id().equals(repositoryName)) {
```

Comment:
Can `repository.id()` be null? shall we swap the values?

Reply:
Default value is empty string if there is no entry in the Settings.toml

Reply:
yeh. since `repositoryName` is already null checked.